### ISSUE TITLE
[MSHARED-480] Use maven-site-plugin's site.xml to use site's skin instead of default when run as mojo

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-reporting-impl#108]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MSHARED-480](https://issues.apache.org/jira/browse/MSHARED-480?redirect=false)** and commented

currently, when a reporting mojo is not used as maven-site-plugin's report but as direct mojo invocation, maven-site-renderer default skin is always used: would be more consistent if using site decoration consistently with maven-site-plugin
Or more precisely, I didn't currently find any reporting plugin that did the effort to mimic maven-site-plugin's decoration model/skin: every reporting plugin I know of uses features of `maven-reporting-impl` for site rendring (with this known limitation)

implementing the feature in `maven-reporting-impl` will make it accessible to any reporting plugin once it upgrades it dependency version

once done, we can activate skin resources copy, that was commented while working on MSHARED-120 (and would not give expected result if skin used is not consistent)

---

**Affects:** maven-reporting-impl 2.4

**Attachments:**
- [screenshot-1.png](https://issues.apache.org/jira/secure/attachment/13053568/screenshot-1.png) (_95.18 kB_)
- [screenshot-2.png](https://issues.apache.org/jira/secure/attachment/13053569/screenshot-2.png) (_89.53 kB_)
- [screenshot-3.png](https://issues.apache.org/jira/secure/attachment/13053573/screenshot-3.png) (_110.97 kB_)
- [screenshot-4.png](https://issues.apache.org/jira/secure/attachment/13053574/screenshot-4.png) (_90.82 kB_)

**Issue Links:**
- [MSHARED-1168](https://issues.apache.org/jira/browse/MSHARED-1168) Add support for markup output

- [MSITE-690](https://issues.apache.org/jira/browse/MSITE-690) Add new "process-reports" goal

**Remote Links:**
- [GitHub Pull Request #15
](https://github.com/apache/maven-reporting-impl/pull/15)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

> I do believe that skin-related must be in the default skin and used through decoration model only
> I don't understand this

the current issue is currently against `maven-reporting-impl` that has to mimic maven-site-plugin's decoration model handling in the context of direct mojo invocation, and that currently was just implemented in a basic way.

With the rework done for Doxia Sitetools 1.7 API, I think it's time to improve maven-reporting-impl

And once maven-site-plugin's skin is used for the template part, resources are not an issue: it is natural to address resources in the same work

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Closing as won't fix.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Here is a comparison from my project:
With Maven Site Plugin:
!screenshot-1.png!
Direct invocation after the fix:
!screenshot-2.png!

Don't mind the heading size, this needs to be fixed in Fluido. The menu is correct since the menu is also populated by the Maven Site Plugin sans the reports.
From my opion the menu on the side or top is useless:
!screenshot-3.png!

From Fluido Skin we need to internally set this:

```
34   <custom>
35     <fluidoSkin>
36       <topBarEnabled>false</topBarEnabled>
37       <sideBarEnabled>false</sideBarEnabled>
38     </fluidoSkin>
39   </custom>
```

and the result:
!screenshot-4.png!

Note: Now the context contains the information about standalone execution so we can distinguish.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Let me know what you think.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

another aspect that this now makes clear, and I was struggling to explain my feeling until now (I now see why...): on sections titles and the change to have Doxia section 1 rendered with h1 instead of h2 previously, which seemed crazy

Here, it proves why section 1 to HTML h2 was not crazy: as you can see, there is the title of the page ("Active DNS Locator" in your example) that is rendered in the header section of the HTML output (by the skin...)

i don't know of the skin renders the title as h1 or as another malrkup, but the (unsaid until now) reasoning of the historic choice of rendering section 1 to h2 was to have this title as h1 (in old skins, I suppose).

We'll need to discuss on that again, i suppose, now that I was able to clarify the historical reasoning...

definitely, this new rendering clarifies a lot the page content vs the site "decoration" from Doxia Siterenderer with its skins

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [d2032091c1bd86c6f52d80a23852aff0e8c0b6ae](https://gitbox.apache.org/repos/asf?p=maven-reporting-impl.git;a=commit;h=d2032091c1bd86c6f52d80a23852aff0e8c0b6ae).
